src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/Domain.cs

```diff
@@ -1265,7 +1265,7 @@ private ArrayList GetTrustsHelper(string? targetDomainName)
             {
                 try
                 {
-                    error = UnsafeNativeMethods.DsEnumerateDomainTrustsW(serverName, (int)DS_DOMAINTRUST_FLAG.DS_DOMAIN_IN_FOREST | (int)DS_DOMAINTRUST_FLAG.DS_DOMAIN_DIRECT_OUTBOUND | (int)DS_DOMAINTRUST_FLAG.DS_DOMAIN_DIRECT_INBOUND, out domains, out count);
+                    error = Interop.Netapi32.DsEnumerateDomainTrustsW(serverName, (int)DS_DOMAINTRUST_FLAG.DS_DOMAIN_IN_FOREST | (int)DS_DOMAINTRUST_FLAG.DS_DOMAIN_DIRECT_OUTBOUND | (int)DS_DOMAINTRUST_FLAG.DS_DOMAIN_DIRECT_INBOUND, out domains, out count);
```

Comment:
The parameter types of these methods should be changed to corresponding enums so that casts will be unnecessary.

Reply:
Done in 245b06684255b885c433d8332b9ed6f2cb816dd8

This did lead me down a rabbit hole though, and left me with a few more questions:
 - should the `DS_DOMAINTRUST_FLAG` enum be `: uint` (rather than the default `int` base for enums) since the corresponding method signature [takes a `ULONG` for `Flags`](https://learn.microsoft.com/en-us/windows/win32/api/dsgetdc/nf-dsgetdc-dsenumeratedomaintrustsw)?
 - similarly, there are other places where we cast this enum to an `int` when comparing to `DS_DOMAIN_TRUSTS.Flags` on the managed side, populated by a call to `Interop.Netapi32.DsEnumerateDomainTrustsW` via `out domains` (e.g. https://github.com/dotnet/runtime/blob/b09e18e8b6092e35611c2567bd55e8909ea78d7d/src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/Domain.cs#L1268) - should that `DS_DOMAIN_TRUSTS` class be using `uint`s for fields that correspond to `ULONG`s on [`DS_DOMAIN_TRUSTSW` on the unmanaged side](https://learn.microsoft.com/en-us/windows/win32/api/dsgetdc/ns-dsgetdc-ds_domain_trustsw) too? Should it also be renamed and moved to common interop too?
 - I feel like the `DS_DOMAIN_TRUST_FLAGS` enum should probably also be moved to common interop folder along with some of the others?

I also spent today setting up an Active Directory server and workstation VMs, and ran the tests from the workstation. There were three test failures which appear to be pre-existing to this PR. The test failures are all due to expectations around exception types (`COMExpcetion` vs `DirectoryServicesCOMException`):
 - the first is `TestInvalidSearchFilter` which I think is a badly written test because it intentionally introduces bad credentials (I think it shouldn't, and should instead be using just a bad search filter based on the name, and be expecting an `ArgumentException`?
 - the other two (`TestInvalidUsernameAndPassword` and `TestMissingUserAndPasswordInfo`)are intentionally using bad credentials, but the confusing thing is that if you give a valid username but an invalid password, then the tests pass (`DSCException`), but if you provide a bad username _and_ password_ then `ADsGetLastError` seems to give an error code of 0 and the library doesn't convert the `COMException` and so the test fails. Is this known behaviour?

Sorry if some of these questions have obvious answers - still getting used to interop code - thanks again for your time!

Reply:
I've pushed a few commits which try to express my intent with the above questions - nothing that can't be reverted though!

Reply:
Those existing test failures are worth an issue if there isn't one already.

These tests were mostly written a long time ago, and in some cases need cleanup; only the ones that don't need a test server are run often. The others require a test server be set up manually. (In the parts of DS that work against Linux we can use docker  - you will have seen the test scripts I created in one of the DS test folders for that. But even for Linux this is done manually)

Reply:
(I meant https://github.com/dotnet/runtime/blob/main/src/libraries/Common/tests/System/DirectoryServices/LDAP.Configuration.xml stuff)

Reply:
> * should the `DS_DOMAINTRUST_FLAG` enum be `: uint`

As always, consistency with existing interop enum definitions should be preferred.

> Should it also be renamed and moved to common interop too?

The enum should be shared with method using the same group of flags.

Reply:
After a bit more searching I came across [this comment](https://github.com/dotnet/runtime/pull/47703#issuecomment-778678410) from your previous investigation into these test failures which I think outlines the situation as it still currently stands:

> TestInvalidSearchFilter (which passes a bad password) expect a DirectoryServicesCOMException but are getting a COMException.

As I said above, I think this test should _not_ be passing a bad password, but rather valid credentials, and then be looking for a `ArgumentException` when using the invalid filter in a search.

> TestMissingUserAndPasswordInfo expects to get an exception when enumerating the directory, but it does not.

I think the reason it isn't, is because we both ran the test on an AD-connected machine under an AD user, so the bind is inheriting from the currently logged in user... Running it on a non-AD-connected machine against OpenJD (still with no creds) passes because then it has no "ambient" auth and really does have an exception thrown.

>  TestInvalidUserAndPassword expect a DirectoryServicesCOMException but are getting a COMException. this is because AdGetLastError is returning no error after the failed bind. I do not know why.

Yeah, I came to the same conclusion for this re. `ADsGetLastError` - when the username is correct but the password is wrong, I get no error from it. When the username is wrong, `ADsGetLastError` gives back what I expect. No idea on that one.

> TestComInterfaces fails because it expects the security descriptor of a new OU it creates to correspond to the user it's bound as but instead it's BUILTIN. This does not occur on .NET Framework. I do not know why.

At least on this branch, that test doesn't run against .NET Framework (possibly due to an `ItemGroup` condition change last year?), but I get the failure against both .NET Framework and non-Framework, and I think it's because it's not checking the OU it created, but the DirectoryEntry above it instead, so it should be a simple fix.

Shall I open this up as a new issue tomorrow, so that all the info from the linked comment by you and what I've put above are in one place?